test/DynamicExpresso.UnitTest/ConstructorTest.cs

```diff
@@ -179,6 +179,84 @@ public void Array_multi_dimension_constructor()
 			Assert.Throws<ParseException>(() => target.Parse("new int[,] { { 1 }, { 2 } }"));
 		}
 
+		[Test]
+		public void Ctor_NewDictionaryWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.Dictionary<,>));
+			var l = target.Eval<System.Collections.Generic.Dictionary<int, string>>("new Dictionary<int, string>(){{1, \"1\"}, {2, \"2\"}, {3, \"3\"}, {4, \"4\"}, {5, \"5\"}}");
+			Assert.AreEqual(5, l.Count);
+			for (int i = 0; i < l.Count; ++i)
+			{
+				Assert.AreEqual(i + 1 + "", l[i + 1]);
+			}
+		}
+
+		[Test]
+		public void Ctor_NewMyClassWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			var l = target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{\"6\" },7	}.Add(true)");
+			Assert.AreEqual(5, l.MyArr.Length);
+			for (int i = 0; i < l.MyArr.Length; ++i)
+			{
+				Assert.AreEqual(i + 1, l.MyArr[i]);
+			}
+			Assert.AreEqual("6", l.StrProp);
+			Assert.AreEqual(7, l.IntField);
+		}
+
+
+		[Test]
+		public void Ctor_NewMyClassWithCross()
+		{
+			string StrProp = "";
+			var mc = new MyClassAdder()
+			{
+				{1,2,3 },
+				{StrProp = "6" }
+			};
+			if (StrProp == "6")
+			{
+
+			}
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			Assert.DoesNotThrow(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7}", new Parameter("StrProp", "0")));
+			Assert.DoesNotThrow(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},StrProp = \"6\",7}", new Parameter("StrProp", "0")));
+			Assert.DoesNotThrow(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},string.Empty, 7}"));
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7	}"));
+		}
+		[Test]
+		public void Ctor_NewListWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.List<>));
+			var intList = target.Eval<System.Collections.Generic.List<int>>("new List<int>(){1, 2, 3, 4, 5}");
+			Assert.AreEqual(5, intList.Count);
+			for (int i = 0; i < intList.Count; ++i)
+			{
+				Assert.AreEqual(i + 1, intList[i]);
+			}
+		}
+
+		[Test]
+		public void Ctor_NewListWithString()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.List<>));
+			var list = target.Eval<System.Collections.Generic.List<string>>("new List<string>(){string.Empty}");
+			Assert.AreEqual(1, list.Count);
+			for (int i = 0; i < list.Count; ++i)
+			{
+				Assert.AreSame(string.Empty, list[i]);
+			}
+			Assert.DoesNotThrow(() => target.Eval<System.Collections.Generic.List<string>>("new List<string>(){StrProp = string.Empty}", new Parameter("StrProp", "0")));
+			Assert.DoesNotThrow(() => target.Eval<System.Collections.Generic.List<string>>("new List<string>(){StrValue()}", new Parameter("StrValue", new Func<string>(() => "Func"))));
```

Comment:
Please check the expected values.

Reply:
Added more positive and negative testing around generic list creation.